src/providers/hammer/__tests__/hammer.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { async, TestBed, ComponentFixture } from '@angular/core/testing';
+import { HammerProvider } from '../hammer';
+import { CompetencyComponent } from '../../../pages/test-report/components/competency/competency';
+
+describe('HammerProvider', () => {
+  let fixture: ComponentFixture<CompetencyComponent>;
+  let hammerProvider: HammerProvider;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CompetencyComponent,
+      ],
+      providers: [
+        HammerProvider,
+      ],
+    })
+      .compileComponents()
+      .then(() => {
+        hammerProvider = TestBed.get(HammerProvider);
+        fixture = TestBed.createComponent(CompetencyComponent);
+      });
+  }));
+
+  describe('addPressAndHoldEvent', () => {
+    it('should invoke the provided callback when "pressAndHold" is emitted', () => {
+      const { elementRef } = fixture;
+      hammerProvider.init(elementRef);
+      const postHoldCallback = jasmine.createSpy('someCallback');
+      hammerProvider.addPressAndHoldEvent(postHoldCallback);
+
+      hammerProvider.hammerManager.emit('pressAndHold', {});
```

Comment:
Nice

Reply:
Meh, kind of annoyed that I couldn't get the emulation of press+hold working, but hey-ho.